# Discharge Flowrate

In [1]:
import numpy as np

## Burst Disc Discharge Capacity

The burst disc has the discharge capacity 435 scfm for air.  Calculate the discharge capacity for Ar at room temperature (300K) and 1 atm using ASME Mandatory Appendix 11-1.

$C_p$: heat capacity at constant pressure \
$C_V$: heat capacity at constant volume \
(from [wiki](https://en.wikipedia.org/wiki/Heat_capacity_ratio))

$C_p$ and $C_V$ are taken from the [NIST database](https://webbook.nist.gov/chemistry/fluid/) at 60 F (289 K) and 1 atm

In [2]:
Cp = 20.834
CV = 12.479
k = Cp/CV
k

1.6695248016668003

Calculate C, constant for gas or vapor which is function of the ratio of specific heats, $k = C_p/C_V$, with the formulae in Figure 11-1.

In [3]:
CompliK = np.power((2/(k+1)), ((k+1)/(k-1)))
C = 520*np.sqrt(k*CompliK)
CompliK, C

(0.31622059648141143, 377.8286251461336)

Calculate the discharge capacity for Ar with the formula in 11-1,

$W = CKAP\sqrt{M/T}$,

where

$A$ = actual discharge area of the safety valve, in.$^2$ (mm $^2$)

$C$ = constant for gas or vapor which is function of the ratio of specific heats, $k = C_p/C_V$ (see Figure 11-1)

K = coefficient of discharge [see UG-131(d) and UG-131(e)]

M = molecular weight

P = (set pressure x 1.10) plus atmospheric pressure, psia (MPa $_{abs}$)

T = absolute temperature at inlet [(°F + 460) (K)]

W = flow of any gas or vapor, lb/hr 

In [4]:
# W for air, unit in lb/hr
# In3toFt3 = 0.0005787, 1 in^3 = 0.0005787 ft^3
# Air density (dair) = 0.073841*0.8+0.084387*0.2 (lbm/ft3), where lbm is pound-mass
# Discharge capacity for air = 435 scfm
In3toFt3 = 0.0005787
dair = 0.073841*0.8+0.084387*0.2
# use the unit of in^3/hr
DC_air = 435*60
Wair = DC_air*In3toFt3*dair
Wair

1.1471571373140002

In [5]:
# Cair from ASME App. 11-1
# Mair = molecular weight of the air from ASME App. 11-1
# MAr = molecular weight of Ar from wiki
# WAr = Flow of Ar (lb/hr)
Cair = 356
Mair = 28.97
MAr = 39.962
WAr = Wair*C*np.sqrt(MAr)/(Cair*np.sqrt(Mair))
WAr

1.4299377502434982

In [6]:
# dAr = Ar density = 0.10535 lbm/ft3
dAr = 0.10535
# DC_Ar, convert to in^3/min
DC_Ar = WAr/dAr/In3toFt3/60
DC_Ar

390.910970809926

The burst disc has the discharge capacity of 391 scfm for Ar.

## Initial LAr Filling

At the beginning of the LAr filling, all the LAr will evaporate when contacting the cryostat at room temperature.  Based on our experience, we consumed a 230-L LAr supply dewar not shorter than 1 hour.  At the beginning we opened the dewar smaller than the time when LAr mostly ended with liquid.  Hence an overestimation of the LAr flowrate is 230/60 liter/minute.

In [7]:
# LAr flowrate: liter/minute
LAr_L_min = 230/60

LAr density: 1.396 g/cm^3

In [8]:
# The mass of LAr filling in a minute, unit in g
d = 1.396
LAr_g_min = d*LAr_L_min*1000
LAr_g_min

5351.333333333333

How many moles of Ar entering the vessel in a minute? \
1 mole of Ar has the mass of 39.962g

In [9]:
# How many moles of Ar
M_A = 39.962
Ar_nMole_min = LAr_g_min/M_A
Ar_nMole_min

133.91054835426988

At 14.7 psia and 300 K, 1 mole of Ar 24.595 has the volume of 24.595 liter

In [10]:
# Ar volume in liter
V_Mole = 24.595
ArFlow_L = Ar_nMole_min*V_Mole
# Ar volume in ft^3
ArFlow_cft = ArFlow_L*0.035
ArFlow_L, ArFlow_cft

(3293.5299367732673, 115.27354778706437)

At the initial LAr filling, all the LAr evaporates, creating Ar gas flow of 3293.5 splm = 115.27 scfm.  It is smaller than the discharge capacity of the burst disc.

## Failure Mode During Operation

1. Thermosyphon fails (no cooling power)
2. The vacuum jacket of the dewar fails (air or Ar in the vacuum jacket)

### Thermosyphon Failure

Thermosyphon evaporator outer radius = 1" = 2.54cm

Thermosyphon evaporator wall thickness, L = (2-1.88)/2" = 0.06" = 0.15cm

Thermosyphon evaporator inner radius = 1.88/2" = 0.94" = 2.39cm

$T_L$ = nitrogen temperature in the TS evaporator = 85.051K at 2.3 bar (NIST)

$T_H$ = Ar temperature near the TS evaporator = 293K

Stainless steel 304 conductivity at 100K, k ~ 9.34W/mK (Thermal Conductivity by T. Ashworth and D. Smith)

Ar Latent Heat of Evaporation at boiling point, HB = 163000 J/kg

Ar volume at 87.305K and 14.7 psia (boiling point), DenB = 0.17316 m3/kg

Ar volume at 293K and 14.7 psia, DenSTP = 0.60127 m3/kg

How much liquid nitrogen in the TS?

Nitrogen volume at 293K, 1bar = 24.356 L/mol

Liquid Nitrogen volume at 85.051K, 2.3bar (inside the TS evaporator) = 0.036387 L/mol

The upper bound of the needed nitrogen amount = 70L

70L nitrogen in the TS evaporator = 0.105L = 105 cm^3

In [11]:
0.036387*70/24.356

0.10457751683363443

TS evaporator inner area = $2.39^2\pi cm^2$ = 17.95cm2

In [12]:
(2.39**2)*np.pi

17.94509139657026

Height of liquid nitrogen in the TS evaporator = 105cm^3/17.95cm^2 = 5.85cm

In [13]:
105/17.95

5.84958217270195

Heat transfer

$P = k\frac{A}{L}(T_H-T_L)$

In [14]:
A = 2.54*2*np.pi*105/17.95
A

93.35518225207775

In [15]:
# conductivity of SS 304 in W/mK
k = 9.34
# L unit in cm
L = 0.15
TH = 293
TL = 85.051

In [16]:
P = k*A*(TH-TL)*0.001/L
P

1208.7900723816172

TS will provide 1209W cooling power?!  I think it is also limited to the condenser.  I will use 1kW anyway.

In [17]:
# W
P = 1000
# J/kg
HB = 163000
# m3/kg
DenSTP = 0.60127
# L/s
ArFlow_L_sec = P/HB*DenSTP*1000
# L/min
ArFlow_L_min = ArFlow_L_sec*60
ArFlow_L_sec, ArFlow_L_min

(3.6887730061349693, 221.32638036809817)

Ar flow rate is 221.33 slpm = 7.82 scfm

### Vacuum Jacket Failure

Dewar bottom area:

$A_b = \int^{\sin\theta = 1/2}_{0} 2\pi r^2 \sin\theta d\theta = 2\pi r^2(1-\frac{\sqrt{3}}{2})$

The bottom curvature radius, r = 14"

$r\sin\theta = $ 7"

Dewar bottom heat transfer length:

$L_b = r-r\cos\theta_1$, where $r\sin\theta_1$ = 8

In [18]:
# cm
r = 14*2.54
# cm^2
Ab = 2*np.pi*r**2*(1-np.sqrt(3)/2)
# cm
Lb = r - r*np.sqrt(1-16/49)
Ab, Lb

(1064.4513810121991, 6.377621755586812)

Wall area, Aw = $2\pi R H_{dewar}$, where R = 7", $H_{dewar}$ = 24"

Wall area, Aw = 6810.15 cm^2

Wall heat transfer length, Lw = 1" = 2.54 cm

In [19]:
# cm
Rdewar = 7*2.54
# cm
Hdewar = 24*2.54
# cm^2
Aw = 2*np.pi*Rdewar*Hdewar
Aw

6810.14851907037

Vacuum jacket fails and is full of air:

air conductivity at 293K, kair = 25.56mW/mK

Ar temperature at 16.7 psia, TAr = 88.531K

In [20]:
# cm
Lw = 2.54
# miniW/mK
kair = 25.56
# K
TAr = 88.531
# W/mK
P_air = kair*(TH-TAr)*(Ab/Lb+Aw/Lw)*0.01*0.001
P_air

148.84635812599973

The failed vacuum jacket filled with air will introduce 149W of heat

In [21]:
P_air/HB*DenSTP*1000*60

32.94362567500117

All this 149W of heat will boil the LAr corresponding to 33 slpm = 1.17 scfm

Vacuum jacket fails and is full of Ar:

Ar conductivity at 293K, kAr = 0.0172 W/mK

In [44]:
# W/mk
kAr = 0.0172
# W/mk
P_Ar = kAr*(TH-TAr)*(Ab/Lb+Aw/Lw)*0.01
P_Ar

100.16265100810624

The failed vacuum jacket filled with Ar will introduce 100W of heat

In [45]:
P_Ar/HB*DenSTP*1000*60

22.168636995697188

All this 100W of heat will boil the LAr corresponding to 22 slpm = 0.78 scfm